src/peft/tuners/waveft/waverec2.py

```diff
@@ -0,0 +1,286 @@
+import functools
+import typing
+import warnings
+from collections.abc import Callable, Sequence
+from functools import partial
+from typing import Any, List, Literal, NamedTuple, Optional, Protocol, Union, cast, overload
+
+import numpy as np
+import pywt
```

Comment:
This package is not a PEFT dependency, so this import will fail. I don't know what package exactly that is, maybe [this one](https://pypi.org/project/waveletnn/), but we very rarely add new external dependencies, only if absolutely necessary, so I think we should remove it.